src/main/java/com/productdock/library/inventory/application/service/UpdateBookStatusService.java

```diff
@@ -0,0 +1,32 @@
+package com.productdock.library.inventory.application.service;
+
+import com.productdock.library.inventory.adapter.out.kafka.messages.BookAvailabilityMessage;
+import com.productdock.library.inventory.application.port.in.UpdateBookStatusUseCase;
+import com.productdock.library.inventory.application.port.out.messaging.BookAvailabilityMessagingOutPort;
+import com.productdock.library.inventory.application.port.out.persistence.InventoryRecordsPersistenceOutPort;
+import com.productdock.library.inventory.domain.RentalRecord;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@AllArgsConstructor
+public class UpdateBookStatusService implements UpdateBookStatusUseCase {
```

Comment:
Maybe it's better to name it `UpdateBookStateService`

Reply:
Considering we already use "Status" wording in rentalStatus, similar should be applied here